src/transformations.jl

```diff
@@ -0,0 +1,569 @@
+##################
+## LLA <-> ECEF ##
+##################
+
+immutable LLAfromECEF{Datum} <: AbstractTransformation{LLA, ECEF}
+    a::Float64      # major axis
+    f::Float64      # flattening
+    e2::Float64     # Eccentricity squared
+    e2m::Float64    # 1 - e2m
+    e2a::Float64    # |e2m|
+    e4a::Float64    # e2m^2
+
+    datum::Datum
+
+    function LLAfromECEF(a,f,e2,e2m,e2a,e4a,datum)
+        if !(isfinite(a) && a > 0)
+            error("Major radius is not positive")
+        end
+        if !(isfinite(f) && f < 1)
+            error("Minor radius is not positive")
+        end
+        return new(a,f,e2,e2m,e2a,e4a,datum)
+    end
+end
+Base.show(io::IO, trans::LLAfromECEF) = print(io, "LLAfromECEF($(trans.datum))")
+
+"""
+    LLAfromECEF(datum)
+
+Construct a `AbstractTransformation` object to convert from ECEF coordinates
+to LLA coordinates. Pre-caches ellipsoidal parameters for efficiency.
+"""
+function LLAfromECEF{Datum}(datum::Datum)
+    el = ellipsoid(datum)
+
+    a = el.a
+    b = el.b
+    f = 1 - b/a
+    e2 = f*(2-f) # or el.e²
+    e2m = (1-f)*(1-f)  #1 - e2
+    e2a = abs(e2)
+    e4a = e2*e2
+
+    return LLAfromECEF{Datum}(a, f, e2, e2m, e2a, e4a, datum)
+end
+
+
+function transform(trans::LLAfromECEF, ecef::ECEF)
```

Comment:
I'm not going to try to review this, I'm sure the tests will catch any porting errors much more efficiently than me :-)